andela/technites-bn-backend
# #168050001 Reset password via email

#### What does this PR do?
This PR contains Reset password via email endpoint, this allows user to request password reset and receive the link via email.


#### Description of Task to be completed?

- Create SendGrid account and API

- Seach email provided if it exists and sign token

- Send email with link

- Decode token from received link

-Reset password

#### How should this be manually tested?
There are two ways:

- The first one is using the swagger documentation via `/api-docs` then this below

1. Run `git checkout reset-password-via-email-168050001` to your terminal .

2. Run `npm install` 

3. Remember to add the variable `SENDGRID_API_KEY` to your `.env` file with the corresponding value. 

4. Run `npm run dev`

5. run `sequelize db:migrate` to create PasswordResets table and models

6. make sure you already have a registered user

7. To request email reset link  use this route `/api/v1/auth/reset` using `POST` method, you will also need to add in an `email` field.

8. Check your email and copy the link to postman using `PATCH` method and pass in `password` and `confirm_password` fields.

10. You will receive a response after it is complete

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168050001](https://www.pivotaltracker.com/story/show/168050001)
#### What are the relevant Github Issues?
N/A
#### Screenshots (if appropriate)
<img width="1122" alt="Screen Shot 2019-09-10 at 16 36 04" src="https://user-images.githubusercontent.com/42216593/64623988-3d8d7700-d3ea-11e9-8780-677c7b071f45.png">
<img width="1118" alt="Screen Shot 2019-09-10 at 16 36 19" src="https://user-images.githubusercontent.com/42216593/64623989-3d8d7700-d3ea-11e9-888b-7687283e2d72.png">
<img width="812" alt="Screen Shot 2019-09-10 at 16 38 56" src="https://user-images.githubusercontent.com/42216593/64623992-3e260d80-d3ea-11e9-98a7-1d329423ba72.png">


#### Questions:
N/A
